<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
    <class>DisciplineForm</class>
    <widget class="QWidget" name="DisciplineForm">
        <layout class="QVBoxLayout" name="mainLayout">

            <!-- ===================== Название дисциплины ===================== -->
            <item>
                <layout class="QHBoxLayout" name="infoLayout">
                    <item>
                        <widget class="QLabel" name="labelDiscipline">
                            <property name="text">
                                <string>Название дисциплины:</string>
                            </property>
                        </widget>
                    </item>
                    <item>
                        <widget class="QLineEdit" name="discipline_name"/>
                    </item>
                </layout>
            </item>

            <!-- ===================== Часы дисциплины ===================== -->
            <item>
                <layout class="QGridLayout" name="hoursLayout">

                    <item row="0" column="0">
                        <widget class="QLabel" name="labelTotal">
                            <property name="text">
                                <string>Общие часы:</string>
                            </property>
                        </widget>
                    </item>
                    <item row="0" column="1">
                        <widget class="QSpinBox" name="total_hours">
                            <property name="maximum">
                                <number>10000</number>
                            </property>
                        </widget>
                    </item>

                    <item row="0" column="2">
                        <widget class="QLabel" name="labelLecture">
                            <property name="text">
                                <string>Лекции:</string>
                            </property>
                        </widget>
                    </item>
                    <item row="0" column="3">
                        <widget class="QSpinBox" name="lecture_hours">
                            <property name="maximum">
                                <number>10000</number>
                            </property>
                        </widget>
                    </item>

                    <item row="1" column="0">
                        <widget class="QLabel" name="labelPractice">
                            <property name="text">
                                <string>Практики:</string>
                            </property>
                        </widget>
                    </item>
                    <item row="1" column="1">
                        <widget class="QSpinBox" name="practice_hours">
                            <property name="maximum">
                                <number>10000</number>
                            </property>
                        </widget>
                    </item>

                    <item row="1" column="2">
                        <widget class="QLabel" name="labelSelfStudy">
                            <property name="text">
                                <string>Сам.раб.:</string>
                            </property>
                        </widget>
                    </item>
                    <item row="1" column="3">
                        <widget class="QSpinBox" name="self_study_hours">
                            <property name="maximum">
                                <number>10000</number>
                            </property>
                        </widget>
                    </item>

                    <!-- СДО -->
                    <item row="2" column="0">
                        <widget class="QLabel" name="labelSDO">
                            <property name="text">
                                <string>СДО часы:</string>
                            </property>
                        </widget>
                    </item>
                    <item row="2" column="1">
                        <widget class="QSpinBox" name="sdo_hours">
                            <property name="maximum">
                                <number>10000</number>
                            </property>
                        </widget>
                    </item>

                    <item row="2" column="2" colspan="2">
                        <widget class="QCheckBox" name="sdo_auto_checkbox">
                            <property name="text">
                                <string>Автоматически (25% от общих часов)</string>
                            </property>
                            <property name="checked">
                                <bool>true</bool>
                            </property>
                        </widget>
                    </item>

                </layout>
            </item>

            <!-- ===================== Настройка семестров ===================== -->
            <item>
                <widget class="QGroupBox" name="sem_manage_group">
                    <property name="title">
                        <string>Настройка семестров</string>
                    </property>

                    <layout class="QHBoxLayout" name="sem_manage_layout">
                        <item>
                            <widget class="QLabel" name="labelFirstSem">
                                <property name="text">
                                    <string>Первый семестр:</string>
                                </property>
                            </widget>
                        </item>
                        <item>
                            <widget class="QSpinBox" name="first_semester_spin">
                                <property name="minimum">
                                    <number>1</number>
                                </property>
                                <property name="maximum">
                                    <number>20</number>
                                </property>
                                <property name="value">
                                    <number>1</number>
                                </property>
                            </widget>
                        </item>

                        <item>
                            <widget class="QLabel" name="labelSemCount">
                                <property name="text">
                                    <string>Количество семестров:</string>
                                </property>
                            </widget>
                        </item>
                        <item>
                            <widget class="QSpinBox" name="semesters_count_spin">
                                <property name="minimum">
                                    <number>1</number>
                                </property>
                                <property name="maximum">
                                    <number>12</number>
                                </property>
                                <property name="value">
                                    <number>1</number>
                                </property>
                            </widget>
                        </item>

                        <item>
                            <widget class="QPushButton" name="apply_semesters_btn">
                                <property name="text">
                                    <string>Применить семестры</string>
                                </property>
                            </widget>
                        </item>
                    </layout>

                </widget>
            </item>

            <!-- ===================== План часов по семестрам ===================== -->
            <item>
                <widget class="QGroupBox" name="semester_group">
                    <property name="title">
                        <string>План часов по семестрам (введите по типам)</string>
                    </property>

                    <layout class="QVBoxLayout" name="semester_layout">
                        <item>
                            <layout class="QVBoxLayout" name="semester_container"/>
                        </item>
                    </layout>

                </widget>
            </item>

            <!-- ===================== Summary по семестрам ===================== -->
            <item>
                <widget class="QGroupBox" name="semester_summary_group">
                    <property name="title">
                        <string>Использование часов по семестрам</string>
                    </property>

                    <layout class="QVBoxLayout" name="semester_summary_layout"/>
                </widget>
            </item>

            <!-- ===================== Разделы ===================== -->
            <item>
                <widget class="QPushButton" name="add_section_btn">
                    <property name="text">
                        <string>+ Добавить раздел</string>
                    </property>
                </widget>
            </item>

            <item>
                <widget class="QScrollArea" name="sections_area">
                    <property name="widgetResizable">
                        <bool>true</bool>
                    </property>
                    <widget class="QWidget" name="sections_widget">
                        <layout class="QVBoxLayout" name="sections_container"/>
                    </widget>
                </widget>
            </item>

            <!-- ===================== Status ===================== -->
            <item>
                <widget class="QLabel" name="status_label">
                    <property name="text">
                        <string></string>
                    </property>
                </widget>
            </item>

            <!-- ===================== Save ===================== -->
            <item>
                <widget class="QPushButton" name="save_btn">
                    <property name="text">
                        <string>Сохранить программу</string>
                    </property>
                </widget>
            </item>

        </layout>
    </widget>

    <resources/>
    <connections/>
</ui>
